src/players/controllers/players.controller.ts

```diff
@@ -57,8 +59,16 @@ export class PlayersController {
   @Get(':id')
   @UseInterceptors(ClassSerializerInterceptor)
   @UseFilters(DocumentNotFoundFilter)
-  async getPlayer(@Param('id', ObjectIdValidationPipe) playerId: string) {
-    return this.playersService.getById(playerId);
+  async getPlayer(
+    @Param('id', ObjectIdOrSteamIdPipe) playerId: ObjectIdOrSteamId,
+  ) {
+    switch (playerId.type) {
+      case 'object-id':
+        return this.playersService.getById(playerId.objectId);
+
+      case 'steam-id':
+        return this.playersService.findBySteamId(playerId.steamId64);
```

Comment:
nit: you should await these calls for async stack traces

Reply:
Ty, fixed